### PR TITLE
Maintenance windows - action schedule react component

### PR DESCRIPTION
**Pull request by hustodemon**
This is the React counterpart of the struts PR:
https://github.com/uyuni-project/uyuni/pull/2265

Enhanced the `ActionSchedule` component with the maintenance windows picker.

Checked all `ActionSchedule` usages and made them maintenance-window-aware.

Review commit-by-commit. The first commit is a squashed & rebased version of https://github.com/hustodemon/uyuni/pull/1.

### TODO
- [x] fix ssm-subscribe-channels.js: pass systemIds correctly
- [x] address cluster usages
- [x] image-build.js: does it need maintenance windows? i think in general no, but our code should be flexible enough to handle this if the action type `image.build` changes in the future
- [x] looks like the code hides the date picker in case there are maintenance windows. didn't we agree on greying out? In the end we will hide that component (I reverted the Struts patch)
- [x] deuglify the state management in `action-schedule.js`
- [x] handle multiple schedules
- [ ] decide: guest-properties-form.js: which type of action and on which system id? there are multiple actions.
- [x] spin this off: test running actions with virtualization, clusters and image building

## GUI diff

* maint. windows picker

ss-maint

* situation, when systems have multiple schedules (this can happen in ssm)

ss-multi

- [x] **DONE**

## Documentation
https://github.com/SUSE/spacewalk/issues/11254

- [x] **DONE**

## Test coverage
- No tests: We do not have tests for react components

- [x] **DONE**

## Links

Tracks https://github.com/SUSE/spacewalk/issues/11230

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: web/html/src/components/action-schedule.js [hunk 1]
```diff
@@ -190,43 +263,13 @@ class ActionSchedule extends React.Component<ActionScheduleProps, ActionSchedule
...
-                  <Combobox id="action-chain" name="action_chain" selectedId={this.state.actionChain.id}
-                            data={this.state.actionChains} onSelect={this.onSelectActionChain}
-                            onFocus={this.onFocusActionChain} />}
-              </div>
-            </div>
+            { 
+              this.renderPickers()
+            }
+            {
+              this.state.actionChains && this.state.actionChain && this.renderActionChainPicker()
```
Comment (ncounter) [inline web/html/src/components/action-schedule.js:291]: Isn't it enough to check for the single `this.state.actionChain`? IIUC if it is present there will always be a `this.state.actionChains` array (in some case an `Array(1)`, but still). Am I missing something?

File: web/html/src/components/action-schedule.js [hunk 2]
```diff
@@ -55,31 +55,26 @@ class ActionSchedule extends React.Component<ActionScheduleProps, ActionSchedule
...
-        earliest: props.earliest,
-        isMaintenanceModeEnabled: false,
-        maintenanceWindow: {},
-        maintenanceWindows: [],
-        systemIds: props.systemIds ? props.systemIds : [],
-        actionType: props.actionType ? props.actionType : "",
       };
     }
 
+    this.state = Object.assign(commonState, actionChainsState);
```
[the remark this answers is not in the document]
Comment (hustodemon) [reply web/html/src/components/action-schedule.js:79]: Fixing.

File: web/html/src/components/action-schedule.js (hunk 1 above)
Comment (hustodemon) [reply web/html/src/components/action-schedule.js:290]: I'd rather not touch this code in this PR. This was only meant as a refactoring and I used condition from the original code.

**Comment by hustodemon** [comment]
Merged yesterday. Thanks for your review @ncounter !

**Pull request merged**